src/prefect/tasks/kubernetes/pod.py

```diff
@@ -571,3 +573,118 @@ def run(
         api_client.replace_namespaced_pod(
             name=pod_name, namespace=namespace, body=body, **kube_kwargs
         )
+
+
+class ReadNamespacedPodLogs(Task):
+    """
+    Task for reading logs from a namespaced pod on Kubernetes. Logs can be streamed by
+    providing a `func_stream` function which then will be called for each log line. If
+    `func_stream` = `None`, the task returns all logs for the pod until that point.
+
+    Note that all initialization arguments can optionally be provided or overwritten at runtime.
+
+    This task will attempt to connect to a Kubernetes cluster in three steps with
+    the first successful connection attempt becoming the mode of communication with a
+    cluster.
+
+    1. Attempt to use a Prefect Secret that contains a Kubernetes API Key. If
+    `kubernetes_api_key_secret` = `None` then it will attempt the next two connection
+    methods. By default the value is `KUBERNETES_API_KEY` so providing `None` acts as
+    an override for the remote connection.
+    2. Attempt in-cluster connection (will only work when running on a Pod in a cluster)
+    3. Attempt out-of-cluster connection using the default location for a kube config file
+
+    Args:
+        - pod_name (str, optional): The name of a pod to replace
+        - namespace (str, optional): The Kubernetes namespace to patch this pod in,
+            defaults to the `default` namespace
+        - func_stream (Callable, optional): Stream the logs to provided function,
+            for each line the function is called.
```

Comment:
I'd find the following kwarg name (and docstring) clearer. Same in the `run` method below.

```
        - on_log_entry (Callable, optional): If provided, will stream the pod logs
            calling the callback for every line (and the task returns `None`). If not
            provided, the current pod logs will be returned immediately from the task.
```